packages/@o3r/core/schematics/application/index.ts

```diff
@@ -0,0 +1,56 @@
+import { apply, chain, externalSchematic, MergeStrategy, mergeWith, move, renameTemplateFiles, Rule, schematic, strings, template, url } from '@angular-devkit/schematics';
+import { getPackagesBaseRootFolder, getWorkspaceConfig } from '@o3r/schematics';
+import * as path from 'node:path';
+import { NgGenerateApplicationSchema } from './schema';
+import type { PackageJson } from 'type-fest';
+
+/**
+ * Add an Otter application to a monorepo
+ *
+ * @param options Schematic options
+ */
+export function generateApplication(options: NgGenerateApplicationSchema): Rule {
+  const cleanName = strings.dasherize(options.name);
+
+  const createAppPackageJson = (targetPath: string, rootDependencies: Record<string, string | undefined>) => {
+
+    return mergeWith(apply(url('./templates'), [
+      template({
+        ...options,
+        name: cleanName,
+        rootDependencies
+      }),
+      move(targetPath),
+      renameTemplateFiles()
+    ]), MergeStrategy.Overwrite);
+  };
+
+  const ngCliUpdate: Rule = (tree, context) => {
+    const config = getWorkspaceConfig(tree);
+    if (!config) {
+      throw new Error('No workspace configuration file found');
+    }
+    const defaultRoot = getPackagesBaseRootFolder(tree, context, config, 'application');
+
+    /** Path to the folder where to generate the new application */
+    const targetPath = path.posix.resolve('/', options.path || defaultRoot, cleanName);
+    const extendedOptions = { ...options, targetPath, name: cleanName, projectName: cleanName };
+
+    const rootPackageJson = tree.readJson('./package.json') as PackageJson;
+
+    const rootDependencies = { ...rootPackageJson.dependencies, ...rootPackageJson.devDependencies };
```

Comment:
can add `peerDependencies` as well (especially that there is no `devDependency` in the `package.json` generated)

Reply:
I don't think we have peerDeps at root level, do we?